x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -781,9 +797,10 @@ void populateAndRead(
             }
         }
 
-        private void fillGaps(RangeMissingHandler writer, Executor executor, SharedBytes.IO fileChannel, List<SparseFileTracker.Gap> gaps) {
+        private void fillGaps(RangeMissingHandler writer, String executor, SharedBytes.IO fileChannel, List<SparseFileTracker.Gap> gaps) {
+            final var exec = threadPool.executor(executor);
```

Comment:
Can we just do this lookup once up-front, hold the relevant ones in fields on `SharedBlobCacheService`, which is then accessible via the `CacheFile` so we can just pass the actual `Executor` all the way down the stack?